script.js

```diff
@@ -225,101 +231,165 @@ const generateBoard = () => {
+      alert(
+        `No, the person doesnt have ${value}! Remove all people with ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
+  } else if (attribute === 'hair') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} hair! Keep all people with ${value} hair`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
+    } else {
+      alert(
+        `No, the person doesnt have ${value}! Remove all people with ${value}`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+    }
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  } else if (attribute === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all people with ${value} eyes`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+   } else {
+    alert(
+      `No, the person doesnt have ${value} eyes! Remove all people with ${value} eyes`
+    )
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+   }
+  }
+
+  generateBoard();
+
+  console.log('filter-characters')
+};
 
-  // Invoke a function to redraw the board with the remaining people.
-}
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  const confirmedAnswer = confirm('Do you want to make a guess?')
+  if (confirmedAnswer === true)
+  checkMyGuess(personToConfirm)
+  
+  console.log('person to confirm')
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if(personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `
+    <p>Cool! You won the game!</p>`
+  } else {
+    winOrLoseText.innerHTML = `
+    <p>Game over!</p>`
+  }
+  winOrLose.style.display = 'flex'
+  board.style.display = 'none'
+
+  console.log('winner or loser')
+
+};
+
+const playingAgain = () => {
+  board.style.display = 'flex'
+  winOrLose.style.display = 'none'
+  start()
+
+  console.log('play again')
+};
 
 // Invokes the start function when website is loaded
-start()
+start();
```

Comment:
I noticed in your code that some have semicolon while others dont.
Recommend to keep it consistant.